src/Input/Input.tsx

```diff
@@ -0,0 +1,128 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React, { useState } from "react";
+import * as typography from "../typography";
+import * as colors from "../colors";
+import { InputWrapper } from "./InputWrapper";
+
+interface Props {
+  /**
+   * Callback to call when the input text is changed
+   */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+  /**
+   * Defaults to standard
+   */
+  size?: "small" | "standard" | "large";
+
+  placeholder?: string;
+
+  /**
+   * Prefilled value of the input
+   */
+  initialValue?: string;
+
+  /**
+   * If true, the user won't be able to interact with any input
+   */
+  disabled?: boolean;
+
+  /**
+   * Visible description
+   */
+  description?: React.ReactNode;
+
+  /**
+   * Visible label
+   */
+  label?: React.ReactNode;
+
+  /**
+   * If this exists, it will be displayed as an error message
+   * below the input
+   */
+  error?: React.ReactNode;
+
+  /**
+   * If this exists, it will be displayed as an info message
+   * below the input
+   */
+  info?: React.ReactNode;
+
+  /**
+   * an icon to have at the left of the input box, a search for example
+   */
+  icon?: React.ReactNode;
+}
+
+export function Input({
+  label,
+  description,
+  size = "standard",
+  onChange,
+  placeholder,
+  initialValue,
+  error,
+  info,
+  disabled,
+  icon,
+  ...otherProps
+}: Props): ReturnType<React.FC> {
+  const [inputtedText, setInputtedText] = useState(initialValue);
+
+  return (
+    <InputWrapper
+      description={description}
+      label={label}
+      size={size}
+      error={error}
+      info={info}
+      icon={icon}
+    >
+      <input
+        {...otherProps}
+        css={{
+          position: "relative",
+          width: "100%",
+          flex: 1,
+          marginRight: icon ? -30 : 0,
+          paddingLeft: icon ? 30 : size === "small" ? 8 : 12,
+          paddingRight: size === "small" ? 8 : 12,
+          margin: 0,
+          height: size === "standard" ? 36 : size === "small" ? 28 : 42,
+          ...typography.base.base,
+          backgroundColor: disabled ? colors.silver.light : colors.white,
+          border: "solid 1px",
+          borderRadius: 4,
+          borderColor: error ? colors.red.base : colors.silver.darker,
+          ":focus, &[data-force-focus-state]": {
+            borderColor:
+              !disabled && !error
+                ? colors.blue.light
+                : error
+                ? colors.red.base
+                : colors.silver.darker,
+            outline: "none",
+          },
+          ":hover,  &[data-force-hover-state]": {
+            borderColor:
+              !disabled && !error
+                ? colors.grey.light
+                : error
+                ? colors.red.base
+                : colors.silver.darker,
+          },
+        }}
+        type="text"
```

Comment:
'text' is the default type and doesn't need to be passed  
having this here after `{...otherProps}` means the type can't be overridden to number/date/email